config/config_helpers.go

```diff
@@ -242,6 +250,19 @@ func runCommand(args []string, include *IncludeConfig, terragruntOptions *option
 
 	currentPath := filepath.Dir(terragruntOptions.TerragruntConfigPath)
 
+	// caching key based on path and arguments
```

Comment:
NIT: Extract the cache look up logic into a reusable function. The point of the function is less about code reuse and more about helping maintainers understand why we cache in the first place. To make that super clear, add a comment to the function to explain that we use a cache to ensure that any given `run_cmd` is only executed once, even if we have multiple parsing passes through this code, and include a link to https://github.com/gruntwork-io/terragrunt/issues/1427 for further info.